tools/inventory-downloader/src/main.go

```diff
@@ -0,0 +1,211 @@
+var targetPrefix = flag.String("prefix", "storage-advisor/inventory", "Prefix to save files")
+var brokers = flag.String("brokers", "localhost:29092", "kafka brokers")
+var groupId = flag.String("group", "inventory-downloader-gburg-1", "kafka listener group")
+var topic = flag.String("topic", "", "kafka listener group")
+
+type Data struct {
+	ProjectId int
+	Payload   InventoryManifest
+}
+
+func processPayload(cfg aws.Config, ctx context.Context, inputManifest InventoryManifest, projectId int) error {
+	var err error
+	outputManifest := InventoryManifest{PartitionDate: inputManifest.PartitionDate}
+
+	for i, file := range inputManifest.Files {
+		filename, err := extractFilenameFromURL(file.Url)
+		if err != nil {
+			return fmt.Errorf("Error getting filename form URL: %w", err)
+		}
+
+		objectKey := fmt.Sprintf("%s/%s/%s/%s/%s/%s",
+			*targetPrefix,
+			strconv.Itoa(projectId),
+			inputManifest.Bucket,
+			inputManifest.PartitionDate,
+			"data",
+			filename)
+		log.Printf("(%d/%d) Downloading file to s3://%s/%s\n", i, len(inputManifest.Files), *bucketName, objectKey)
+		err = processFile(cfg, ctx, objectKey, file.Url)
+		if err != nil {
+			return fmt.Errorf("Error processing file: %w", err)
+		}
+		outputManifest.Files = append(outputManifest.Files, InventoryFile{Key: objectKey, MD5checksum: file.MD5checksum})
+	}
+
+	manifestKey := fmt.Sprintf("%s/%s/%s/%s/%s",
+		*targetPrefix,
+		strconv.Itoa(projectId),
+		inputManifest.Bucket,
+		inputManifest.PartitionDate,
+		"manifest.json")
+	manifestBytes, err := json.Marshal(outputManifest)
+	if err != nil {
+		return fmt.Errorf("Error Marshalling manifest: %w", err)
+	}
+
+	err = uploadToS3(cfg, ctx, bytes.NewReader(manifestBytes), *bucketName, manifestKey)
+	if err != nil {
+		return fmt.Errorf("Error uploading file to S3: %w", err)
+	}
+	log.Printf("Finished processing manifest and uploaded s3://%s/%s\n", *bucketName, manifestKey)
+
+	return nil
+}
+
+func main() {
+	flag.Parse()
+
+	ctx := context.TODO()
+	cfg, err := config.LoadDefaultConfig(ctx)
+	cfg.Region = "eu-central-1"
+	if err != nil {
+		log.Println("Error:", err)
+		return
+	}
+
+	kafkaConfig := &kafka.ConfigMap{
+		"bootstrap.servers":        *brokers,
+		"group.id":                 *groupId,
+		"auto.offset.reset":        "earliest",
+		"go.events.channel.enable": true,
+	}
+
+	consumer, err := kafka.NewConsumer(kafkaConfig)
+	if err != nil {
+		log.Printf("Error creating consumer: %v\n", err)
+		os.Exit(1)
+	}
+
+	topics := []string{*topic}
+	err = consumer.SubscribeTopics(topics, nil)
+	if err != nil {
+		log.Printf("Error subscribing to topics: %v\n", err)
+		os.Exit(1)
+	}
+
+	sigchan := make(chan os.Signal, 1)
+	signal.Notify(sigchan, syscall.SIGINT, syscall.SIGTERM)
+
+	log.Printf("Listening to topics %s\n", topics)
+
+	run := true
+	for run == true {
+		select {
+		case sig := <-sigchan:
+			log.Printf("Caught signal %v: terminating\n", sig)
+			run = false
```

Comment:
That's more of stylistic comment - but maybe "while true" look with "break" will be a bit cleaner.

Reply:
Done